### PR TITLE
fix: allIsInterface not reseted #790

### PR DESCRIPTION
**Pull request by Grummfy**
fix for #790

### CODE REVIEW

**Comment by Grummfy** [comment]
I'm wondering if it's the same behaviour for eachInstanceIsUnique

**Comment by mageekguy** [comment]
Hi guys!
This behaviour is not a bug.
`allIsInterface` is a global configuration directive for the mock generator, because it is the philosophical opposite of its default behavior.
Default behaviour is not true OOP, because it allow the developper to build code based on implementation.
The `allIsInterface` behaviour "force" the developper to build code based upon true abstraction (aka interface), so it enforce the user to do true OOP.
So, in my opinion, allow the user to mix this two philosophie is not the way to go, and it is the reason why `allIsInterface` is a global configuration directive, instead of a local directive which may be reset after mock instantiation.
Morevover, this PR is a BC break, because tests suite which was configured to use `allIsInterface` will be silently switch to "implementation based" mode.
In conclusion, it's a "no go" for me for this PR.
Have a good day!
